flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/xa/JdbcXaSinkFunction.java

```diff
@@ -286,12 +287,15 @@ private void prepareCurrentTx(long checkpointId) throws IOException {
         outputFormat.flush();
         try {
             xaFacade.endAndPrepare(currentXid);
+            outputFormat.reconnect(false);
```

Comment:
Is this the fix for FLINK-22311?

Reply:
No, the intention here is to associate `outputFormat` with a new connection (I'll add a comment).